src/components/TodoElement/TodoElement.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../../context/TodoContext';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+interface Props {
+  todo: Todo,
+}
+
+export const TodoElement: React.FC<Props> = ({ todo }) => {
+  const { completed, title, id } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(title);
+
+  const {
+    deleteTodoHandler,
+    todosIdToDelete,
+    renameTodoHandler,
+    toggleChangeHandler,
+    setErrorMessage,
+  } = useContext(TodoContext);
+
+  const titleInputRef = useRef<HTMLInputElement | null>(null);
+
+  const shouldDisplayLoader = id === 0 || todosIdToDelete.includes(id);
+
+  const handleTodoDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTodoChange = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    try {
+      if (todoTitle) {
+        await renameTodoHandler(todo, todoTitle);
+      } else {
+        await deleteTodoHandler(id);
+      }
+
+      setIsEditing(false);
+    } catch (error) {
+      if (error === ErrorMessage.UpdateError) {
+        setErrorMessage(ErrorMessage.UpdateError);
+        throw new Error();
```

Comment:
The error text must also be passed to the `Error` class, otherwise it makes no sense.
Fix this in all code.